### PR TITLE
Issue/2801 post list recycler

### LINKED ISSUE #2801: Use RecyclerView for post list
[Calypsofication](https://github.com/wordpress-mobile/WordPress-Android/issues/2793) of the post list requires using cards which can be trashed - meaning some animation will be required. Switching to a RecyclerView will greatly simplify this.

I've made this a separate task from supporting cards in order to keep the PRs smaller.

### PR DESCRIPTION
**Pull request by nbradbury**
Fix #2801 - changes the posts/pages list to use a RecyclerView. Feature-wise nothing has been added - this should work exactly as it did before. The goal is to make this change prior to switching to card views for posts/pages, which will be a separate PR after this one is merged.

### CODE REVIEW

**Comment by tonyr59h** [comment]
When in dual-pane mode I noticed that the list selection can get messed up if you select a post while refreshing. The item you select in the list is highlighted, a toast says it's refreshing, but the preview does not change.

**Comment by nbradbury** [comment]
@tonyr59h Ready for round two!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java [hunk 1]
```diff
@@ -78,6 +77,10 @@ public void onCreate(Bundle savedInstanceState) {
         FragmentManager fm = getFragmentManager();
         mPostList = (PostsListFragment) fm.findFragmentById(R.id.postList);
 
+        // the post list should highlight the selected post when in dual-pane mode
+        boolean isDualPane = (fm.findFragmentById(R.id.postDetail) != null);
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java]: :heart_eyes_cat: for clarifying meaning with a variable.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java [hunk 2]
```diff
@@ -503,14 +414,99 @@ public void onSaveInstanceState(Bundle outState) {
...
+                mPostList.requestPosts(false);
+                mPostList.setRefreshing(true);
+            } else {
+                AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(PostsActivity.this);
+                dialogBuilder.setTitle(getResources().getText(R.string.connection_error));
+                dialogBuilder.setMessage(mErrorMsg);
+                dialogBuilder.setPositiveButton("OK",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int whichButton) {
+                                // Just close the window.
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java]: Passing null will have the same effect here, your call if it's worth a commit or not.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostsActivity.java]: That's existing code, not something I touched in this PR, so I'll just
leave it as is.

**Comment by tonyr59h** [comment]
Sounds good, :shipit:

**Pull request merged**